src/Arcus.Scripting.DevOps/Scripts/Set-AzDevOpsArmOutputsToVariableGroup.ps1

```diff
@@ -0,0 +1,84 @@
+        [string][parameter(Mandatory = $true)]$VariableGroupName,
+        [string][parameter(Mandatory = $true)]$variableName,
+        [string][parameter(Mandatory = $true)]$variableValue
+    )
+    BEGIN
+    {
+        #Retrieve project details
+        Write-Host Retrieving project details
+        
+        [String]$project = "$env:SYSTEM_TEAMPROJECT"
+        [String]$projectUri = "$env:SYSTEM_TEAMFOUNDATIONCOLLECTIONURI"
+        [String]$apiVersion = "4.1-preview.1"
+        
+        Write-Host Project: $project
+        Write-Host ProjectUri: $projectUri
+        
+
+        #Set authorization headers 
+        Write-Host Set authorization headers
+        if ([string]::IsNullOrEmpty($env:SYSTEM_ACCESSTOKEN))
+        {
+            Write-Error "The SYSTEM_ACCESSTOKEN environment variable is empty. Remember to explicitly allow the build job to access the OAuth Token!"
+        }
+        $headers = @{ Authorization = "Bearer $env:SYSTEM_ACCESSTOKEN" }
+
+
+        #Get variable group
+        Write-Host Get variable group
+        $getVariableGroupUrl= $projectUri + $project + "/_apis/distributedtask/variablegroups?api-version=" + $apiVersion + "&groupName=" + $VariableGroupName
+        $variableGroup = (Invoke-RestMethod -Uri $getVariableGroupUrl -Headers $headers -Verbose) 
+        
+        if($variableGroup.value)
+        {
+            #Set properties for update of existing variable group
+            Write-Host Set properties for update of existing variable group
+            $variableGroup = $variableGroup.value[0]
+			$variableGroup | Add-Member -Name "description" -MemberType NoteProperty -Value "Variable group that got auto-updated by release '$env:Release_ReleaseName'." -Force
+            $method = "Put"
+            $upsertVariableGroupUrl = $projectUri + $project + "/_apis/distributedtask/variablegroups/" + $variableGroup.id + "?api-version=" + $apiVersion    
+        }
+        else
+        {
+            #Set properties for creation of new variable group
+            Write-Host Set properties for creation of new variable group
+            $variableGroup = @{name=$VariableGroupName;type="Vsts";description="Variable group that got auto-updated by release '$env:Release_ReleaseName'.";variables=New-Object PSObject;}
+            $method = "Post"
+            $upsertVariableGroupUrl = $projectUri + $project + "/_apis/distributedtask/variablegroups?api-version=" + $apiVersion
+        }
+
+        #Add variable
+        $variableGroup.variables | Add-Member -Name $variableName -MemberType NoteProperty -Value @{value=$variableValue} -Force
+
+        #Upsert variable group
+        Write-Host Upsert variable group
+        $body = $variableGroup | ConvertTo-Json -Depth 10 -Compress
+        Write-Host $body
+        Invoke-RestMethod $upsertVariableGroupUrl -Method $method -Body $body -Headers $headers -ContentType 'application/json; charset=utf-8' -Verbose
+    }
+}
+
+$armOutputs = ConvertFrom-Json $env:ArmOutputs
```

Comment:
The expected environment name containing the ARM outputs, should be added to the documentation as well.

Reply:
Yeah, I was look for that. Either we do this like this and _hide_ it in the script but document it; or we allow consumers to pass along the ARM outputs or the name of the environment variable where it's located?

Reply:
Ideally, we would allow uses the specify the name of the environment variable themselves, so it's not fixed, but have this as a default.
However, I tried something similar in the past, but passed along the environment variable name, didn't work for me at that time (or I was missing something)

Reply:
If the PS command doesn't work that way, we can always use the .NET variant in that case.

Reply:
Not quite sure if I'm following what you mean?
I'd try to stick to PS as much as possible here, if not possible, let's go for a default name and add clear documentation.
Perhaps later on we can look into a different approach in order to dynamically set the environment variable name.

Reply:
I mean this

```powershell
[System.Environment]:SetEnvironmentVariable
```

Reply:
Updated it like proposed, feel free to take a look.

Reply:
Must have overlooked the existence of that method :)
Nice find! Perfect solution.